src/components/implementation/capmgr/naive/mem_mgr.c

```diff
@@ -25,41 +25,41 @@ memmgr_heap_page_allocn(unsigned int npages)
 	return dst_pg;
 }
 
-int
-memmgr_shared_page_allocn_cserialized(vaddr_t *pgaddr, int *unused, int npages)
+cbuf_t
+memmgr_shared_page_allocn_cserialized(vaddr_t *pgaddr, int *unused, u32_t npages)
 {
 	spdid_t cur = cos_inv_token();
 	struct cap_comp_info  *cur_rci = cap_info_comp_find(cur);
 	struct cap_shmem_info *cur_shi = cap_info_shmem_info(cur_rci);
-	int shmidx = -1;
+	cbuf_t shmid = 0;
 
 	if (!cur_rci || !cap_info_init_check(cur_rci)) goto done;
 	if (!cur_shi) goto done;
 
-	shmidx = cap_shmem_region_alloc(cur_shi, npages);
-	if (shmidx < 0) goto done;
+	shmid = cap_shmem_region_alloc(cur_shi, npages);
+	if (!shmid) goto done;
 
-	*pgaddr = cap_shmem_region_vaddr(cur_shi, shmidx);
+	*pgaddr = cap_shmem_region_vaddr(cur_shi, shmid);
 
 done:
-	return shmidx;
+	return shmid;
 }
 
-int
-memmgr_shared_page_map_cserialized(vaddr_t *pgaddr, int *unused, int idx)
+u32_t
```

Comment:
Little strange the `shared_mapn` still returns an `int`.  What do you think about this being a `size_t` instead?

I'm just having a little trouble understanding why `u32_t` is the right type for this.